File: ET3ALEM_CLIENT_WEB/src/app/question/Models/short-answer-attempt.ts
import { QuizQuestion } from "src/app/quiz/Model/quizQuestion";
import { QuestionAttempt } from "./question-attempt";

export class ShortAnswerAttempt extends QuestionAttempt {
    Answer: string;

    constructor(Id: number = 0, QuizQuestion: QuizQuestion = null, Grade: number = 0, Answer: string = '') {
        super(Id, QuizQuestion, Grade);
        this.Answer = Answer;
    }
}

File: ET3ALEM_SERVER/BusinessEntities/Models/QuestionAttempt/ShortAnswerAttempt.cs
#nullable enable
using System;

namespace BusinessEntities.Models
{
    public class ShortAnswerAttempt : QuestionAttempt
    {
        public string? Answer { get; set; }

        public override void GradeQuestion()
        {
            var question = QuizQuestion.Question as ShortAnswerQuestion;

            if (string.IsNullOrWhiteSpace(question?.PossibleAnswers))
            {
                IsGraded = false;
                return;
            }

            IsGraded = true;
            foreach (string possibleAnswer in question.PossibleAnswers.Split(','))
            {
                var compareType = question.CaseSensitive
                    ? StringComparison.InvariantCulture
                    : StringComparison.InvariantCultureIgnoreCase;
                //check if the answers match the possible answer
                var isMatching = Answer != null && Answer.Trim()
                    .Equals(possibleAnswer.Trim(), compareType);

                //if it's matching, give full grade and return
                if (isMatching)
                {
                    Grade = QuizQuestion.Grade;
                    return;
                }
            }

            //if none match set Grade to zero
            Grade = 0;
        }
    }
}